rclcpp/src/rclcpp/publisher_base.cpp

```diff
@@ -264,7 +274,8 @@ PublisherBase::setup_intra_process(
   const rcl_publisher_options_t & intra_process_options)
 {
   // Intraprocess configuration is not allowed with "durability" qos policy non "volatile".
-  if (this->get_actual_qos().durability != RMW_QOS_POLICY_DURABILITY_VOLATILE) {
+  auto actual_durability = this->get_actual_qos().get_rmw_qos_profile().durability;
```

Comment:
Separate enhancement idea; it might be nice if `QoS` had a getter `rmw_qos_durability_policy_t durability() const`

Reply:
Yeah, I considered adding those now, but I wanted the pr to be small (accessors could be backported, this couldn't). It will add a lot of bloat to the QoS class but maybe it's worth it. However, I imagine cases like this are rare, maybe the only other use case is printing the QoS, but I think adding them is a good idea in general.